f5/bigip/tm/gtm/server.py

```diff
@@ -0,0 +1,75 @@
+# coding=utf-8
+#
+#  Copyright 2014-2016 F5 Networks Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+"""BIG-IP® Global Traffic Manager™ (GTM®) pool module.
+
+REST URI
+    ``http://localhost/mgmt/tm/gtm/server``
+
+GUI Path
+    ``DNS --> GSLB : Servers``
+
+REST Kind
+    ``tm:gtm:server:*``
+"""
+
+from f5.bigip.resource import Collection
+from f5.bigip.resource import Resource
+
+
+class Servers(Collection):
+    """BIG-IP® GTM server collection"""
+    def __init__(self, gtm):
+        super(Servers, self).__init__(gtm)
+        self._meta_data['allowed_lazy_attributes'] = [Server]
+        self._meta_data['attribute_registry'] = \
+            {'tm:gtm:server:serverstate': Server}
+
+
+class Server(Resource):
+    """BIG-IP® GTM server resource"""
+    def __init__(self, servers):
+        super(Server, self).__init__(servers)
+        self._meta_data['required_json_kind'] = 'tm:gtm:server:serverstate'
+        self._meta_data['required_creation_parameters'].update(
+            ('datacenter', 'addresses'))
+        self._meta_data['attribute_registry'] = {
+            'tm:gtm:server:virtual-servers:virtual-serverscollectionstate':
+                Virtual_Servers_s
+        }
+
+
+class Virtual_Servers_s(Collection):
+    """BIG-IP® GTM virtual server sub-collection"""
+    def __init__(self, server):
+        super(Virtual_Servers_s, self).__init__(server)
+        self._meta_data['allowed_lazy_attributes'] = [Virtual_Servers]
+        self._meta_data['required_json_kind'] = \
+            'tm:gtm:server:virtual-servers:virtual-serverscollectionstate'
+        self._meta_data['attribute_registry'] = {
+            'tm:gtm:server:virtual-servers:virtual-serversstate':
+                Virtual_Servers}
+
+
+class Virtual_Servers(Resource):
```

Comment:
I realize I let this slide in the last PR, but we have a precedent in the sys/snmp code whereby the resource being operated on in singular. So methinks we should continue that here and fix the other PR.

See here
https://github.com/F5Networks/f5-common-python/blob/development/f5/bigip/tm/sys/snmp.py#L64